cluster-autoscaler/core/scaledown/pdb/pdb_test.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pdb
+
+import (
+	"fmt"
+	"testing"
+
+	apiv1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	. "k8s.io/autoscaler/cluster-autoscaler/utils/test"
+)
+
+func TestCanDisrupt(t *testing.T) {
+	one := intstr.FromInt(1)
+	label1 := "label-1"
+	label2 := "label-2"
+	pdb1 := &policyv1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: "ns",
+		},
+		Spec: policyv1.PodDisruptionBudgetSpec{
+			MinAvailable: &one,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					label1: "true",
+				},
+			},
+		},
+		Status: policyv1.PodDisruptionBudgetStatus{
+			DisruptionsAllowed: 1,
+		},
+	}
+	pdb2 := &policyv1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "bar",
+			Namespace: "ns",
+		},
+		Spec: policyv1.PodDisruptionBudgetSpec{
+			MinAvailable: &one,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					label2: "true",
+				},
+			},
+		},
+		Status: policyv1.PodDisruptionBudgetStatus{
+			DisruptionsAllowed: 2,
+		},
+	}
+	testCases := []struct {
+		name            string
+		podsLabel1      int
+		podsLabel2      int
+		pdbs            []*policyv1.PodDisruptionBudget
+		pdbsDisruptions [2]int32
+		canDisrupt      bool
+	}{
+		{
+			name:       "No pdbs",
+			podsLabel1: 2,
+			podsLabel2: 1,
+			canDisrupt: true,
+		},
+		{
+			name:            "Not enough pod disruption budgets",
+			podsLabel1:      2,
+			podsLabel2:      1,
+			pdbs:            []*policyv1.PodDisruptionBudget{pdb1, pdb2},
+			pdbsDisruptions: [2]int32{1, 2},
+			canDisrupt:      false,
+		},
+		{
+			name:            "Enough pod disruption budgets",
+			podsLabel1:      2,
+			podsLabel2:      3,
+			pdbs:            []*policyv1.PodDisruptionBudget{pdb1, pdb2},
+			pdbsDisruptions: [2]int32{2, 4},
+			canDisrupt:      true,
+		},
+	}
+	for _, test := range testCases {
+		pdb1.Status.DisruptionsAllowed = test.pdbsDisruptions[0]
+		pdb2.Status.DisruptionsAllowed = test.pdbsDisruptions[1]
+		pdbRemainingDisruptions := NewPdbRemainingDisruptions(test.pdbs)
+		pods := []*apiv1.Pod{}
+		for i := 0; i < test.podsLabel1; i++ {
```

Comment:
I'd extract this to some `makePodsWithLabel(string, int)` helper func to avoid having two essentially identical loops.